src/Sparrow/Collections/ConcurrentSet.cs

```diff
@@ -98,5 +99,21 @@ public override string ToString()
         {
             return Count.ToString("#,#", CultureInfo.InvariantCulture);
         }
+
+        public void UnionWith(IEnumerable<T> other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
+
+            foreach (T item in other)
+            {
+                if (_inner.TryAdd(item, null))
+                {
+                    Interlocked.Increment(ref _innerCountForEmpty);
+                }
```

Comment:
why not simply use  `TryAdd` ?

Reply:
`.Count` is locking

Reply:
I mean the `TryAdd` of the `ConcurrentSet` class

Reply:
@garayx if you handle it the way that Karmel suggested in the test then you don't have to introduce this method at all

Reply:
are there any draw backs of introducing this method?

Reply:
Introducing a new public method to the API of `ConcurrentSet` means that it can be used in the future in other places (not just in tests). So there is no drawback of introducing this method, but then it would be nice to have tests which verify its correctness.

Reply:
I literally use it to confirm tests results:
https://github.com/ravendb/ravendb/blob/ac9ffd513581d7f2b0b626ffb8a9519579c22154/test/SlowTests/Client/Subscriptions/ConcurrentSubscriptionsTests.cs#L1373

But I have added tests which verify its correctness

Reply:
@karmeli87 done